### ISSUE TITLE
[java] SimplifyBooleanExpressions false-negative on boolean ternary expression

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2585]
**Affects PMD Version:** 
6.22.0

**Rule:** [SimplifyBooleanExpressions](https://pmd.github.io/latest/pmd_rules_java_design.html#simplifybooleanexpressions)

**Description:**
It is recommended to have SimplifyBooleanExpressions additionally check the case like <code>bar = isFoo() ? true : false</code>
Similar to <code>boolean bar = (isFoo() == true)</code>, <code>bar = isFoo() ? true : false</code> can also be simplified to <code>bar = isFoo()</code>

**Code Sample demonstrating the suggestion:**
Example 1:
```java
// can be simplified to "capitals = caps.isSelected()"
capitals = caps.isSelected() ? true:false;
```

Example 2:
```java
// can be simplified to "Current_Token == Token.TOK_BOOL_TRUE"
Current_Token == Token.TOK_BOOL_TRUE ? true : false;
```

**Type:**
Suggestions for rule improvement

**Running PMD through:**
CLI

**Comment by adangel** [comment]
I think we should close this issue as "wont-fix", because all the cases are for [SimplifiedTernary](https://pmd.github.io/latest/pmd_rules_java_design.html#simplifiedternary) and not [SimplifyBooleanExpressions](https://pmd.github.io/latest/pmd_rules_java_design.html#simplifybooleanexpressions). Meaning: we won't enhance SimplifyBooleanExpressions to detect simplifiable ternaries...

**Comment by adangel** [comment]
Closing this as won't fix - SimplifiedTernary will cover this case, see #3603
